libgadget/petaio.h

```diff
@@ -55,20 +67,21 @@ petaio_build_selection(int * selection,
  * IO_REG_WRONLY declares an io block which is written, but is not read on snapshot load.
  * */
 #define IO_REG(name, dtype, items, ptype) \
```

Comment:
Perhaps worth it to plumb IOTable up through these macros -- a bit of search and replacement, but avoids an hidden argument.

Reply:
Yes, but it is a fair bit of careful search and replace. I don't really want to just now.

Reply:
I did it anyway :-)